src/internal/internalPane.ts

```diff
@@ -1 +1,255 @@
+      const view = context.session.paneRegistry.byName('internal')
+      return (
+        view &&
+        view.predicates &&
+        !!(typeof view.predicates[pred.uri] !== 'undefined')
+      )
+    }
+
+    var div = dom.createElement('div')
+    div.setAttribute('class', 'internalPane')
+    div.setAttribute(
+      'style',
+      'background-color: #ddddff; padding: 0.5em; border-radius: 1em;'
+    )
+
+    function deleteRecursive (kb: IndexedFormula, folder: NamedNode) {
+      const fetcher = (kb as any).fetcher
+      if (!fetcher) {
+        console.error('No fetcher available')
+        return
+      }
+      return new Promise(function (resolve, reject) {
+        fetcher.load(folder).then(function () {
+          const promises = kb.each(folder, ns.ldp('contains')).map(file => {
+            if (kb.holds(file, ns.rdf('type'), ns.ldp('BasicContainer'))) {
+              return deleteRecursive(kb, file as NamedNode)
+            } else {
+              console.log('deleteRecursive leaf file: ' + file)
+              return fetcher.webOperation('DELETE', (file as NamedNode).uri)
+            }
+          })
+          Promise.all(promises).then(
+            () => {
+              console.log('deleteRecursive empty folder: ' + folder)
+              fetcher
+                .webOperation('DELETE', folder.uri)
+                .then(() => {
+                  console.log('Deleted Ok: ' + folder)
+                  resolve()
+                })
+                .catch((err: string) => {
+                  var str = 'Unable to delete ' + folder + ': ' + err
+                  console.log(str)
+                  reject(new Error(str))
+                })
+              resolve()
+            },
+            err => {
+              alert(err)
+              reject(err)
+            }
+          )
+        })
+      })
+    }
+
+    const isDocument = subject.uri && !subject.uri.includes('#')
+    if (isDocument) {
+      const controls = div.appendChild(dom.createElement('table'))
+      controls.style.width = '100%'
+      controls.style.margin = '1em'
+      const controlRow = controls.appendChild(dom.createElement('tr'))
+
+      const deleteCell = controlRow.appendChild(dom.createElement('td'))
+      const isFolder =
+        (subject.uri && subject.uri.endsWith('/')) ||
+        store.holds(subject, ns.rdf('type'), ns.ldp('Container'))
+      const noun = isFolder ? 'folder' : 'file'
+      if (!isProtectedUri(subject)) {
+        console.log(subject)
+        var deleteButton = widgets.deleteButtonWithCheck(
+          dom,
+          deleteCell,
+          noun,
+          function () {
+            if (
+              !confirm(
+                `Are you sure you want to delete ${subject}? This cannot be undone.`
+              )
+            ) {
+              return
+            }
+            // @@ TODO Remove casing of store.fetcher
```

Comment:
To remove  a casting of store.fetcher and store.updater, how about making superclasses of Store, like ConnectedStore or OnlineStore for the one which has a fetcher, and LiveStore superclass of OnlineStore which would also have an updater ?

Reply:
Or is it too valuable to be able to make store and attach its helper objects later?

Reply:
Was thinking the same thing, and have a local commit that's about to be a PR for that. Can push it tomorrow.

Reply:
Have pushed a PR to `pane-registry` with proposed changes - https://github.com/solid/pane-registry/pull/5